src/types/observablearray.ts

```diff
@@ -124,8 +133,14 @@ class ObservableArrayAdministration<T> implements IInterceptable<IArrayWillChang
 		if (oldLength !== this.lastKnownLength)
 			throw new Error("[mobx] Modification exception: the internal structure of an observable array was changed. Did you use peek() to change it?");
 		this.lastKnownLength += delta;
-		if (delta > 0 && oldLength + delta > OBSERVABLE_ARRAY_BUFFER_SIZE)
-			reserveArrayBuffer(oldLength + delta);
+		if (delta > 0 && oldLength + delta + 1 > OBSERVABLE_ARRAY_BUFFER_SIZE)
+			reserveArrayBuffer(oldLength + delta + 1);
+		if (delta > 0)
```

Comment:
Btw just let me know if you want npm publish rights.

Op vr 8 jul. 2016 om 11:22 schreef Andy Kogut <notifications@github.com>:

> In src/types/observablearray.ts
> <https://github.com/mobxjs/mobx/pull/396#discussion_r70047465>:
>
> > @@ -124,8 +133,14 @@ class ObservableArrayAdministration<T> implements IInterceptable<IArrayWillChang
> >  		if (oldLength !== this.lastKnownLength)
> >  			throw new Error("[mobx] Modification exception: the internal structure of an observable array was changed. Did you use peek() to change it?");
> >  		this.lastKnownLength += delta;
> > -		if (delta > 0 && oldLength + delta > OBSERVABLE_ARRAY_BUFFER_SIZE)
> > -			reserveArrayBuffer(oldLength + delta);
> > +		if (delta > 0 && oldLength + delta + 1 > OBSERVABLE_ARRAY_BUFFER_SIZE)
> > +			reserveArrayBuffer(oldLength + delta + 1);
> > +		if (delta > 0)
>
> @mweststrate <https://github.com/mweststrate>, Yes, thats fine
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mobxjs/mobx/pull/396/files/7565600e293c9a584fc3fd937e45131126f9d747#r70047465>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABvGhHCdLmxLw9LzwA2_ixkSan7-RsRpks5qThbegaJpZM4JHiRJ>
> .
>